### PR TITLE
[PORT] Fix security camera console not following moving borgs

### PR DESCRIPTION
**Pull request by Sarchutar**
## About The Pull Request

Port of: tgstation/tgstation#52767 tgstation/tgstation#52830

Updates the camera in the camera console if the camera/cyborg moves.

I tested this on a local server and it seems to work like intended.

Original description:

> Camera consoles don't tracking moving cameras.
> 
> If the camera's turf changes between ui_updates, the camera now follows.
> 
> It's not perfect, but it gets the job done for what is ostensibly a niche feature and basically feels like any other tracking code.

## Why It's Good For The Game

Security camera consoles now track cyborgs correctly.

## Changelog
:cl: Timberpoes
fix: The security camera console will now correctly track when viewing moving cyborg cameras.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/camera.dm [hunk 1]
```diff
@@ -121,27 +124,46 @@
...
+	if(last_camera_turf == newturf)
+		return
+
+	// Cameras that get here are moving, and are likely attached to some moving atom such as cyborgs.
+	last_camera_turf = get_turf(cam_location)
+
+	var/list/visible_things = active_camera.isXRay() ? range(active_camera.view_range, cam_location) : view(active_camera.view_range, cam_location)
+
+	for(var/turf/visible_turf in visible_things)
+		visible_turfs += visible_turf
```
Comment (park66665) [inline code/game/machinery/computer/camera.dm]:
```suggestion
	if(active_camera.isXRay())
		visible_turfs += RANGE_TURFS(active_camera.view_range, cam_location)
	else
		for(var/turf/T in view(active_camera.view_range, cam_location))
			visible_turfs += T
```

File: code/game/machinery/computer/camera.dm [hunk 2]
```diff
@@ -121,27 +124,46 @@
...
+	// Show static if can't use the camera
+	if(!active_camera?.can_use())
+		show_camera_static()
+		return
 
-		return TRUE
+	var/list/visible_turfs = list()
+
+	// Is this camera located in or attached to a living thing? If so, assume the camera's loc is the living thing.
+	var/cam_location = isliving(active_camera.loc) ? active_camera.loc : active_camera
```
Comment (park66665) [inline code/game/machinery/computer/camera.dm]:
```suggestion
	var/atom/cam_location = isliving(active_camera.loc) ? active_camera.loc : active_camera
```

This, by the way, won't work with item locs; e.g. #3730

**Comment by PowerfulBacon** [comment]
Huh how does this even work, does cameras moving call security console updates or something?

**Comment by park66665** [comment]
SStgui calls `ui_interact` every tick, `ui_interact` calls `update_active_camera_screen` every tick, which checks if the camera has moved and updates visible turfs accordingly. In contrast, the code right now only updates when camera is changed.

**Comment by PowerfulBacon** [comment]
oh yea auto update

**Review by park66665 — DISMISSED** [review]
All clear. This PR is good to go.

**Review by park66665 — APPROVED** [review]
Porting the remaining pieces of tgstation/tgstation#52767 can be done at a later date. All clear. This PR is good to go.

**Pull request merged**